gno.land/pkg/integration/testing_integration.go

```diff
@@ -0,0 +1,257 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"hash/crc32"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/node"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/jaekwon/testify/require"
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+// XXX: This should be centralize somewhere.
+const (
+	test1Addr = "g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5"
+	test1Seed = "source bonus chronic canvas draft south burst lottery vacant surface solve popular case indicate oppose farm nothing bullet exhibit title speed wink action roast"
+)
+
+type testNode struct {
+	*node.Node
+	logger      log.Logger
+	nGnoKeyExec uint // Counter for execution of gnokey.
+}
+
+func SetupGnolandTestScript(t *testing.T, txtarDir string) testscript.Params {
+	t.Helper()
+
+	cmd := exec.Command("go", "list", "-m", "-mod=mod", "-f", "{{.Dir}}", "github.com/gnolang/gno")
+	out, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+
+	// `gnoRootDir` should point to the local location of the gno repository.
+	// It serves as the gno equivalent of GOROOT.
+	gnoRootDir := strings.TrimSpace(string(out))
+
+	// `gnoHomeDir` should be the local directory where gnokey stores keys.
+	gnoHomeDir := filepath.Join(t.TempDir(), "gno")
+
+	// `gnoDataDir` should refer to the local location where the gnoland node
+	// stores its configuration and data.
+	gnoDataDir := filepath.Join(t.TempDir(), "data")
+
+	var muNodes sync.Mutex
+	nodes := map[string]*testNode{}
+
+	persistWorkDir, _ := strconv.ParseBool(os.Getenv("TESTWORK"))
+	return testscript.Params{
+		TestWork: persistWorkDir,
+		Dir:      txtarDir,
+		Setup: func(env *testscript.Env) error {
+			kb, err := keys.NewKeyBaseFromDir(gnoHomeDir)
+			if err != nil {
+				return err
+			}
+
+			// XXX: Add a command to add custom account.
+			kb.CreateAccount("test1", test1Seed, "", "", 0, 0)
+			env.Setenv("USER_SEED_test1", test1Seed)
+			env.Setenv("USER_ADDR_test1", test1Addr)
+
+			env.Setenv("GNOROOT", gnoRootDir)
+			env.Setenv("GNOHOME", gnoHomeDir)
+			env.Setenv("GNODATA", gnoDataDir)
+
+			return nil
+		},
+		Cmds: map[string]func(ts *testscript.TestScript, neg bool, args []string){
+			"sleep": func(ts *testscript.TestScript, neg bool, args []string) {
+				d := time.Second
+				if len(args) > 0 {
+					var err error
+					if d, err = time.ParseDuration(args[0]); err != nil {
+						ts.Fatalf("unable to parse duration %q: %s", args[1], err)
+					}
+				}
+
+				time.Sleep(d)
+			},
```

Comment:
I am not really in favour of this, it's too error-prone and hardware-dependent. If there is any reason where we have to introduce a time delay (to simulate a situation whereby some given time has passed), then we change the value returned by `time.Now()` (by having a helper instead of calling time.Now() directly), but we definitely don't call Sleep(). If we are waiting for Gno.land genesis to terminate, we add a channel somewhere, or an event-emission system so that we know precisely when gno.land genesis terminates

Would the work required to avoid using time.Sleep be too much for this PR?

Reply:
I want to get ride of sleep for the same reasons you mentioned (error-prone and hardware-dependent) , the only reason I introduce sleep was for the comment above... 

if really needed (if we don't find any good solutions for the previous comment) a more elegant approach should be to introduce an `eventually` prefix command instead of sleep. that will act like [assert.Eventually](https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L1833): 
`eventually gnokey maketx call ...`

Reply:
entirely removed the sleep command, thanks to you 🙏  !